treeherder/auth/backends.py

```diff
@@ -73,6 +73,8 @@ def _get_username_from_userinfo(self, user_info):
             return "email/" + email
         elif "github" in subject:
             return "github/" + email
+        elif "google" in subject:
```

Comment:
Should we be using `.startswith()` rather than checking if it's just a substring? Can there be user provided input at the end of the string?

Reply:
Yeah but we could either:
* use `.startswith('ad|Mozilla-LDAP|')`
* only use `startswith()` on the others
* not use `startswith()`, but adjust the substrings used to include the pipe operator (which would make it harder to fake since not valid email character etc)

Anyway perhaps there is no risk here, just wanted to check given I don't know what possible user provided values can end up in `subject` :-)

CC @djmitche

Reply:
This is a good point.  There's not a security concern here -- login uses a better approach -- but startsWith will be more effective against e.g., ad|Mozilla-LDAP|github@mozilla.com.